quantities/src/main/java/com/decathlon/vitamin/compose/quantities/VitaminQuantities.kt

```diff
@@ -0,0 +1,299 @@
+                value = value,
+                onValueChange = onValueChange,
+                editTextEnabled = editTextEnabled,
+                keyboardOptions = keyboardOptions
+
+            )
+
+            RightButton(onAddClicked = onAddClicked, addEnabled = addEnabled, colors = colors)
+        }
+    }
+}
+
+@Composable
+private fun MiddleTextField(
+    modifier: Modifier,
+    addEnabled: Boolean,
+    subtractEnabled: Boolean,
+    hasQuantityError: Boolean,
+    colors: QuantityColors,
+    value: String,
+    onValueChange: (String?) -> Unit,
+    editTextEnabled: Boolean,
+    keyboardOptions: KeyboardOptions,
+) {
+
+    val focus = remember { mutableStateOf(false) }
+
+    Column(
+        modifier = modifier
+    ) {
+        val dividerEnable =
+            if (addEnabled || subtractEnabled) VtmnStatesEnabled else VtmnStatesDisabled
+
+        Divider(
+            color = colors.borderColor.copy(alpha = dividerEnable)
+        )
+
+        val boxModifier = when {
+            hasQuantityError -> {
+                Modifier
+                    .weight(1.0f)
+                    .border(border = BorderStroke(1.dp, colors.borderColorError))
+            }
+            focus.value -> {
+                Modifier
+                    .weight(1.0f)
+                    .border(border = BorderStroke(1.dp, colors.borderColorActive))
+            }
+            else -> {
+                Modifier.weight(1.0f)
+            }
+        }
+        Box(
+            modifier = boxModifier,
+            contentAlignment = Alignment.Center
+        ) {
+            BasicTextField(
+                modifier = Modifier.onFocusChanged { focusState ->
+                   focus.value = focusState.isFocused
+                },
+                value = value,
+                onValueChange = onValueChange,
+                enabled = editTextEnabled && (addEnabled || subtractEnabled),
+                textStyle = VitaminTheme.typography.text2.copy(
+                    textAlign = TextAlign.Center,
+                    color = colors.textColor
+                ),
+                keyboardOptions = keyboardOptions,
+                singleLine = true
+            )
+        }
+
+        Divider(
+            color = colors.borderColor.copy(alpha = dividerEnable),
+            modifier = Modifier
+        )
+    }
+}
+
+@Composable
+private fun RightButton(onAddClicked: () -> Unit, addEnabled: Boolean, colors: QuantityColors) {
+    Button(
+        onClick = onAddClicked,
+        modifier = Modifier
+            .fillMaxHeight()
+            .width(48.dp),
+        enabled = addEnabled,
+        colors = ButtonDefaults.buttonColors(
+            backgroundColor = colors.backgroundColor,
+            disabledBackgroundColor = colors.backgroundColor.copy(alpha = VtmnStatesDisabled)
+        ),
+        shape = RoundedCornerShape(
+            topStart = 0.dp,
+            topEnd = 4.dp,
+            bottomEnd = 4.dp,
+            bottomStart = 0.dp
```

Comment:
Instead of hardcoded values, we should use the new [radius values](https://github.com/Decathlon/vitamin-compose/blob/a8f5ac45066967301b0982ecf12d8f202499b8a8/foundation/src/main/java/com/decathlon/vitamin/compose/foundation/Shape.kt) integrated by @CorentinStamperDecathlon recently.